### PR TITLE
Audio: Adapt to new `make` pattern + improve tests

### PR DESCRIPTION
**Pull request by davedbase**
- [x] Finalize API design
- [x] Make functional minimal demo
- [x] Port and validate tests
- [x] Update README and package.json

### CODE REVIEW

**Comment by thetarnav** [comment]
Looks good.
The event-listener pkg could reduce the logic even more here.
Also, what about returning a Store, instead of an object of accessors in the new `createAudio`?

**Comment by davedbase** [comment]
> Looks good. The event-listener pkg could reduce the logic even more here. Also, what about returning a Store, instead of an object of accessors in the new `createAudio`?

Yup, good suggestion!

**Comment by thetarnav** [comment]
I've made a [`createStaticStore`](#https://github.com/solidjs-community/solid-primitives/blob/027231ae2981f490740a5f1d1101c9539624a5a6/packages/utils/src/index.ts#L323) for such smaller usecases btw.

**Review by thetarnav — CHANGES_REQUESTED** [review]
- [ ] fix tests
- [x] The docs could use some usage examples / types
It's hard to get an idea of what certain primitives will help you with, by just looking through the readme
- [x] There are few random console.logs in index.ts
- [x] expressions like these: `src instanceof Function ? src() : src` could be replaced with `access(src)`, since you already have utils package installed
- [x] Better types for event handlers
```ts
{
   [K in keyof HTMLMediaElementEventMap]?: (event: HTMLMediaElementEventMap[K]) => void;
}
// instead of 
{ [key: string]: EventListener }
```

**Comment by davedbase** [comment]
@thetarnav I adjusted everything, just that pesky test that I have to track. Sigh.

**Comment by davedbase** [comment]
@thetarnav tests are passing (wooohoo). Last review and we publish. Also there's a nice demo I put together

File: packages/audio/test/index.test.ts [hunk 1]
```diff
@@ -1,83 +1,87 @@
...
-  test("test srcObject reactive value path", async () => {
-    const { player } = createAudioPlayer(() => new MediaSource(), []);
-    expect(player.srcObject).toBeInstanceOf(Object);
-  });
-  test("test player with no handlers", async () => {
-    const { player } = createAudioPlayer(() => new MediaSource());
-    expect(player.srcObject).toBeInstanceOf(Object);
-  });
-});
+const testPath = 'https://github.com/solidjs-community/solid-primitives/blob/audio/packages/audio/dev/sample1.mp3?raw=true';
```
Comment (thetarnav) [inline packages/audio/test/index.test.ts:9]: Better not delete that branch haha :D

**Review by thetarnav — APPROVED** [review]
Alright, the `server.ts` needed some corrections, but I took care of it. Looks good otherwise.

**Comment by davedbase** [comment]
Shoot good point, I changed things last minute. Thanks for adjusting it! Going to merge and publish now.

**Pull request merged**